File: extensions/real-estate/front-end/Web/packages/platform/src/views/Command/Inspections/Zone/Check.js
import { numberUtils, Number } from '@willow/ui'

export default function Check({ check }) {
  if (check.statistics.lastCheckSubmittedEntry == null) {
    return '-'
  }

  if (
    check.type !== 'List' &&
    numberUtils.parse(check.statistics.lastCheckSubmittedEntry) != null
  ) {
    return (
      <Number
        value={check.statistics.lastCheckSubmittedEntry}
        format={`,.${'0'.repeat(check.decimalPlaces ?? 0)}`}
      />
    )
  }

  return check.statistics.lastCheckSubmittedEntry
}

File: extensions/real-estate/back-end/PlatformPortalXL/src/Willow.Workflow/Models/Check.cs
using System;
using System.Collections.Generic;

namespace Willow.Workflow
{
    public class Check
    {
        public Guid Id { get; set; }
        public Guid InspectionId { get; set; }
        public string Name { get; set; }
        public CheckType Type { get; set; }
        public string TypeValue { get; set; }
        public int DecimalPlaces { get; set; }
        public double? MinValue { get; set; }
        public double? MaxValue { get; set; }
        public double Multiplier { get; set; }
        public Guid? DependencyId { get; set; }
        public string DependencyValue { get; set; }
        public DateTime? PauseStartDate { get; set; }
        public DateTime? PauseEndDate { get; set; }
        public bool IsArchived { get; set; }
        public bool CanGenerateInsight { get; set; }
        public List<CheckRecord> CheckRecords { get; set; }
        public CheckRecord LastSubmittedRecord { get; set; }
        public InspectionCheckStatistics  Statistics { get; set; }
    }
}
